CSVSource/src/main/java/io/vantiq/extsrc/CSVSource/CSV.java

```diff
@@ -0,0 +1,303 @@
+        }
+        
+        int maxActiveTasks = 5; 
+        int maxQueuedTasks = 10; 
+
+        if (options.get("maxActiveTasks")!=null)
+        {
+            maxActiveTasks = (int) options.get("maxActiveTasks");
+        }
+        if (options.get("maxQueuedTasks")!=null)
+        {
+            maxQueuedTasks = (int) options.get("maxQueuedTasks");
+        }
+
+        executionPool = new ThreadPoolExecutor(maxActiveTasks, maxActiveTasks, 0l, TimeUnit.MILLISECONDS,
+        new LinkedBlockingQueue<Runnable>(maxQueuedTasks));
+
+        fileFilter = new FilenameFilter(){
+            public boolean accept(File dir, String name) {
+               String lowercaseName = name.toLowerCase();
+
+               if (lowercaseName.endsWith(extenstion) && lowercaseName.startsWith(filePrefix)) {
+                  return true;
+               } else {
+                  return false;
+               }
+            }};
+
+
+
+    }
+
+    public void setupCSV(ExtensionWebSocketClient oClient,String fileFolderPath , String fullFilePath ,Map<String, Object>  config ,boolean asyncProcessing ) throws VantiqCSVException
+    {
+        try 
+        {
+            this.fullFilePath = fullFilePath;
+            this.config = config; 
+            this.oClient = oClient;
+
+            serviceWatcher = FileSystems.getDefault().newWatchService(); 
+            Path path = Paths.get(fileFolderPath);
+
+            path.register(serviceWatcher,ENTRY_CREATE);
+
+            log.info("CSV trying to subscribe to {}",fileFolderPath);
+
+           
+            prepareConfigurationData(); 
+
+            new Thread(() -> processThread(fileFolderPath)).start();
+
+            // working on files already exists in folder. 
+            if (options.get("processExistingFiles") != null)
+            {
+                Object processExistingFiles = options.get("processExistingFiles");
+                if (processExistingFiles instanceof Boolean )
+                if ((boolean)processExistingFiles)
+                {
+                    log.info("Start working on existing file in folder {}", fileFolderPath);
+                    hanldeExistingFiles(fileFolderPath);
+                }
+            }
+
+
+        }   
+        catch (Exception e)     
+        {
+
+            log.error("CSV failed to read  from {}",fullFilePath,e);
+            reportCSVError(e);
+        }
+    }
+
+
+    void executeInPool(String fileFolderPath , String filename)
+    {
+        String fullFileName = String.format("%s/%s",fileFolderPath,filename);
+
+        File path = new File (fileFolderPath);
+        if (fileFilter.accept(path, filename))
+        {
+            executionPool.execute(new Runnable() {
+                @Override
+                public void run() {
+                    log.info("start executing {}",fullFileName);
+                    try {
+                        CSVReader.execute(fullFileName,config,oClient);
```

Comment:
sorry but i failed to understand what is your meaning here  ...

Reply:
The comment was simply that there's no status to check or to see if anything else happened.  That the call to ` CSVReader.execute()` either threw an exception or worked as advertised. Because we then go & delete or rename the file.